src/entrypoints/string-split.d.ts

```diff
@@ -0,0 +1,5 @@
+interface String {
+	split<Separator extends string, Limit extends number>(separator: Separator, limit?: Limit): Limit extends 0 ? []
+		: Separator extends '' ? string[]
```

Comment:
I think this could be dangerous when the type is not inferred by a literal and an empty string is passed as type `string`. For example:

```ts
const foo: string = '';
'bar'.split(foo); // <-- would incorrectly yield [string, ...string[]]
```